routes/index.js

```diff
@@ -50,42 +53,59 @@ router.post('/api/users', function (req, res) {
     console.log('Input field empty')
     return res.status(400).send('Input field empty')
   }
-
   // Checks for duplicate
-  const checkdup = {
-    text: 'SELECT COUNT(email) FROM users WHERE email = $1',
-    values: [email.toLowerCase()]
-  }
-  db.query(checkdup)
+  CheckDup(email)
     .then(result => {
-      if (result.rows[0].count !== 0) {
-        res.status(240)
+      if (result != 0) {
+        // console.log(result)
+        res.status(409).send({message:"exist"})
+      } else {
+
+        // Insert user data
+        const query = {
+          text: 'INSERT INTO users(firstname, lastname, email, secret) VALUES ($1, $2, $3, $4)',
+          values: [firstname.toLowerCase(), lastname.toLowerCase(), email.toLowerCase(), secret]
+        }
+        db.query(query)
+          .then(result => {
+            res.send([firstname, lastname, email, secret])
+          })
+          .catch(err => {
+            res.status(500).json(err)
+          })
       }
     })
+ 
+  
+})
 
-  // Insert user data
-  const query = {
-    text: 'INSERT INTO users(firstname, lastname, email, secret) VALUES ($1, $2, $3, $4)',
-    values: [firstname.toLowerCase(), lastname.toLowerCase(), email.toLowerCase(), secret]
+// Helper function for checking user duplicates using email
+// Goes through database and counts where email matches
+async function CheckDup(email) {
+  const checkdup = {
+    text: 'SELECT COUNT(email) FROM users WHERE email = $1',
+    values: [email.toLowerCase()]
   }
-  db.query(query)
-    .then(result => {
-      res.send([firstname, lastname, email, secret])
-    })
-    .catch(err => {
-      res.status(400).json(err)
-    })
-})
+  return await db.query(checkdup)
+  .then(function(result) {
+    return result.rows[0].count
+  })
+}
+
 
 // Route for getting user data from secret
 router.get('/api/user/:secret', function (req, res) {
   const query = {
-    text: 'Select * FROM users WHERE secret = $1',
+    text: 'SELECT f.user1, users.firstname, users.lastname, f.user2, u.firstname AS friendfname, u.lastname As friendlname \
+            FROM users \
+            LEFT JOIN friendships AS f ON  f.user1 = users.id \
+            LEFT JOIN users as u ON f.user2 = u.id WHERE users.secret = $1',
     values: [req.params.secret]
   }
   db.query(query)
     .then(results => {
-      res.send(results.rows)
+      res.send(results)
+      console.log(results)
     })
```

Comment:
As I said in client/src/AddFriend, don't just return raw SQL data -- format it so that's easy to just use by the client.

(just move the `getfriends` code into here)

Why, you might ask? 
1. We don't want to expose the actual format of our database to the outside world. If we later change the table structure (eg rename a field), we should be able to just change our API route -- it shouldn't require everyone calling the API to change the way they parse the code.
2. Prevents duplication of parsing code, if the API route is used more than once in the client.